src/fontra/views/fontinfo/panel-sources.js

```diff
@@ -1,14 +1,642 @@
+
+    const containerContent = [
+      ...labeledTextInput("Source name:", nameController, "sourceName", {}),
+      ...labeledTextInput("Italic Angle:", nameController, "sourceItalicAngle", {}),
+      html.br(),
+      locationElement,
+    ];
+
+    const warningElement = html.div({
+      id: "warning-text",
+      style: `grid-column: 1 / -1; min-height: 1.5em;`,
+    });
+    containerContent.push(warningElement);
+
+    const contentElement = html.div(
+      {
+        style: `overflow: hidden;
+          white-space: nowrap;
+          display: grid;
+          gap: 0.5em;
+          grid-template-columns: max-content auto;
+          align-items: center;
+          height: 100%;
+          min-height: 0;
+        `,
+      },
+      containerContent
+    );
+
+    return { contentElement, warningElement };
+  }
+}
+
+addStyleSheet(`
+:root {
+  --fontra-ui-font-info-sources-panel-max-list-height: 12em;
+}
+
+.fontra-ui-font-info-sources-panel-source-box {
+  background-color: var(--ui-element-background-color);
+  border-radius: 0.5em;
+  padding: 1em;
+  cursor: pointer;
+  display: grid;
+  grid-template-rows: auto auto;
+  grid-template-columns: max-content max-content max-content max-content auto;
+  grid-row-gap: 0.1em;
+  grid-column-gap: 1em;
+}
+
+.fontra-ui-font-info-sources-panel-column {
+  display: grid;
+  grid-template-columns: minmax(4.5em, max-content) max-content;
+  gap: 0.5em;
+  align-items: start;
+  align-content: start;
+  overflow: scroll;
+}
+
+fontra-ui-font-info-sources-panel-source-box.min-height,
+.fontra-ui-font-info-sources-panel-column.min-height {
+  height: 45px;
+}
+
+.fontra-ui-font-info-sources-panel-vertical-metrics {
+  grid-template-columns: minmax(4.5em, max-content) 4em 4em;
+}
+
+.fontra-ui-font-info-sources-panel-header {
+  font-weight: bold;
+}
+
+.fontra-ui-font-info-sources-panel-icon {
+  justify-self: end;
+  align-self: start;
+}
+
+.fontra-ui-font-info-sources-panel-icon.open-close-icon {
+  height: 1.5em;
+  width: 1.5em;
+  transition: 120ms;
+}
+
+.fontra-ui-font-info-sources-panel-icon.open-close-icon.item-closed {
+  transform: rotate(180deg);
+}
+
+`);
+
+class SourceBox extends HTMLElement {
+  constructor(fontAxes, sources, sourceIdentifier, postChange, setupUI) {
+    super();
+    this.classList.add("fontra-ui-font-info-sources-panel-source-box");
+    this.fontAxes = fontAxes;
+    this.sources = sources;
+    this.sourceIdentifier = sourceIdentifier;
+    this.postChange = postChange;
+    this.setupUI = setupUI;
+    this.controllers = {};
+    this.models = this._getModels();
+    this._updateContents();
+  }
+
+  get source() {
+    return this.sources[this.sourceIdentifier];
+  }
+
+  _getModels() {
+    const source = this.source;
+    return {
+      general: {
+        name: source.name,
+        italicAngle: source.italicAngle ? source.italicAngle : 0,
+        isSparse: source.isSparse ? source.isSparse : false,
```

Comment:
Please comment this out for now: this needs to become a more elaborate feature, because a sparse source should not have:
- vertical metrics
- italic angle

We will need to do this as a follow-up.